### PR TITLE
[SCB-887]aysnc servlet timeout is too short and may block container pool when tasks are timeout

### PR DESCRIPTION
**Pull request by liubao68**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java [hunk 1]
```diff
@@ -142,6 +145,18 @@ protected void scheduleInvocation() {
     });
   }
 
+  private boolean isInQueueTimeout() {
+    Object timeout = requestEx.getAttribute(RestConst.REST_REQUEST_IN_QUEUE_TIME);
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java]: 1.direct use invocation.getStartTime()?
is's a nanotime.

2.when invoke from highway, need this logic also

File: transports/transport-highway/src/main/java/org/apache/servicecomb/transport/highway/HighwayConfig.java [hunk 2]
```diff
@@ -18,13 +18,26 @@
...
 import com.netflix.config.DynamicIntProperty;
+import com.netflix.config.DynamicLongProperty;
 import com.netflix.config.DynamicPropertyFactory;
 import com.netflix.config.DynamicStringProperty;
 
 public final class HighwayConfig {
   private HighwayConfig() {
   }
 
+  public static final String KEY_SERVICECOMB_REQUEST_WAIT_IN_POOL_TIMEOUT = "servicecomb.highway.server.requestWaitInPoolTimeout";
```
Comment (wujimin) [inline transports/transport-highway/src/main/java/org/apache/servicecomb/transport/highway/HighwayConfig.java]: maybe no need to 2 different config item for this?

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java [hunk 3]
```diff
@@ -142,6 +145,13 @@ protected void scheduleInvocation() {
     });
   }
 
+  private boolean isInQueueTimeout() {
+    if (System.nanoTime() - invocation.getStartTime() > CommonRestConfig.getRequestWaitInPoolTimeout() * 1_000_000) {
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java]: just return directly

File: transports/transport-rest/transport-rest-servlet/src/main/java/org/apache/servicecomb/transport/rest/servlet/RestAsyncListener.java [hunk 4]
```diff
@@ -63,24 +64,28 @@ public void onTimeout(AsyncEvent event) throws IOException {
     // to avoid concurrent, must lock request
     ServletRequest request = event.getSuppliedRequest();
     HttpServletRequestEx requestEx = (HttpServletRequestEx) request.getAttribute(RestConst.REST_REQUEST);
+    LOGGER.error("Rest request timeout, method {}, path {}.", requestEx.getMethod(), requestEx.getRequestURI());
```
Comment (wujimin) [inline transports/transport-rest/transport-rest-servlet/src/main/java/org/apache/servicecomb/transport/rest/servlet/RestAsyncListener.java]: maybe we can disable async servlet timeout, just control timeout by threadpool timeout?

File: transports/transport-highway/src/main/java/org/apache/servicecomb/transport/highway/HighwayConfig.java (hunk 2 above)
Comment (liubao68) [reply transports/transport-highway/src/main/java/org/apache/servicecomb/transport/highway/HighwayConfig.java]: 2 item is more flexible. I test it for highway this configuration works much better than rest.

File: transports/transport-rest/transport-rest-servlet/src/main/java/org/apache/servicecomb/transport/rest/servlet/RestAsyncListener.java (hunk 4 above)
Comment (liubao68) [reply transports/transport-rest/transport-rest-servlet/src/main/java/org/apache/servicecomb/transport/rest/servlet/RestAsyncListener.java]: I disabled it with -1 by default. And I think it's better to leave timeout setting to users if they really need it in some scenario.

**Comment by liubao68** [comment]
trigger CI

**Pull request merged**